### ISSUE TITLE
New `dotnet_diagnostic.[RuleId].severity` Syntax Fails to Work

### ISSUE THREAD

**Issue by RehanSaeed** [dotnet/roslyn#55542]
**Version Used**: 

- VS 2019, .NET 5, `#error version` = `Compiler version: '3.11.0-4.21403.6 (ae1fff34)'. Language version: latest (9.0).`
- .NET CLI 5.0.400 `#error version` = `Compiler version: '3.11.0-4.21373.6 (bb3bdbbe)'. Language version: 9.0.`.

**Steps to Reproduce**:

The new `dotnet_diagnostic.[RuleId].severity` syntax for specifying severity doesn't seem to work i.e.

## Old Syntax Works
```
# [rule] = [value]:[severity] syntax
csharp_prefer_simple_using_statement = true:suggestion
```

## New Syntax Doesn't Work
```
# [rule].[RuleId].[severity] syntax
csharp_prefer_simple_using_statement = true
dotnet_diagnostic.IDE0063.severity = suggestion
```

My understanding was that the new syntax was added in VS 16.9 and that the old syntax is [due to be deprecated](https://github.com/dotnet/roslyn/issues/44201). See https://github.com/RehanSaeed/EditorConfig/issues/46 for more details.

**Expected Behavior**:

Both the old and new syntaxes should produce the diagnostic with the specified severity.

**Actual Behavior**:

Only the old syntax produces a diagnostic with the severity I set, the new syntax produces a diagnostic with the default severity.

**Comment by RehanSaeed** [comment]
https://github.com/dotnet/roslyn/issues/55541 may be related.

**Comment by sharwell** [comment]
Can you edit the first post to clarify the expected and actual behavior?

**Comment by RehanSaeed** [comment]
@sharwell Updated it, I hope that is now clearer.

**Comment by RehanSaeed** [comment]
Hmmm, I think I got something mixed up. Closing this one.
